bugbug/bugzilla.py

```diff
@@ -93,50 +94,78 @@ def bughandler(bug):
     return all_ids
 
 
-def get(ids_or_query):
-    new_bugs = {}
-
-    def bughandler(bug):
-        bug_id = int(bug["id"])
-
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
-
-        new_bugs[bug_id].update(bug)
-
-    def commenthandler(bug, bug_id):
-        bug_id = int(bug_id)
-
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
+def get(bug_ids):
+    """Function to retrieve Bug Information including history, attachment, comments using Bugzilla REST API.
 
-        new_bugs[bug_id]["comments"] = bug["comments"]
-
-    def attachmenthandler(bug, bug_id):
-        bug_id = int(bug_id)
-
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
-
-        new_bugs[bug_id]["attachments"] = bug
-
-    def historyhandler(bug):
-        bug_id = int(bug["id"])
+    :param bug_ids: find bug information for these `bug_ids`
+    :type bug_ids: list of integers or str or int
+    :return: dict with key as `id`(int) of a bug,  and values as bug_information
+    :rtype: dict
+    """
 
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
+    def filter_keys(array_of_dict_values, required_fields):
+        for a_value in array_of_dict_values:
+            to_be_deleted_fields = set(a_value.keys()).difference(set(required_fields))
+            for extra_field in to_be_deleted_fields:
+                del a_value[extra_field]
+        return array_of_dict_values
 
-        new_bugs[bug_id]["history"] = bug["history"]
+    new_bugs = {}
 
-    Bugzilla(
-        ids_or_query,
-        bughandler=bughandler,
-        commenthandler=commenthandler,
-        comment_include_fields=COMMENT_INCLUDE_FIELDS,
-        attachmenthandler=attachmenthandler,
-        attachment_include_fields=ATTACHMENT_INCLUDE_FIELDS,
-        historyhandler=historyhandler,
-    ).get_data().wait()
+    if isinstance(bug_ids, list):
+        # Expected Format
+        bug_ids_list = list(set(map(int, bug_ids)))
+
+    elif isinstance(bug_ids, six.string_types):
+        bug_ids_list = [int(bug_ids)]
+
+    elif isinstance(bug_ids, int):
+        bug_ids_list = [bug_ids]
+    else:
+        raise ValueError(
+            "Invalid datatype for the parameter-bug_ids with value- {bug_ids} and datatype-{type(bug_ids)}"
+        )
```

Comment:
Instead of the runtime check on type, let's use typing (enforcing the parameter to be a list of IDs and updating all callers to pass a list of IDs).

Reply:
Added  typing enforcing to function - get - Since Set, int, str, are also used in other calls, the typing is enforced to use all these including list.

Reply:
Added typing - enforcing here https://github.com/mozilla/bugbug/pull/1575/commits/cfba88fa234eb3d4d3fcc91eea8a640faec937bc

Reply:
Fixed in this commit : https://github.com/mozilla/bugbug/pull/1575/commits/15b460fe9eb94af905461d114a45860c47c92be8

Reply:
No, this is super-ugly :)
I was suggesting enforcing it is a list and updating all callers to use a list.